### PR TITLE
[Core] Replacing boost::hash_combine

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 1]
```diff
@@ -495,7 +491,7 @@ class ResidualBasedEliminationBuilderAndSolver
     {
         KRATOS_TRY
 
-        // 			boost::timer building_time;
+        //          boost::timer building_time;
```
Comment (RiccardoRossi) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]: we can remove this line

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 2]
```diff
@@ -280,14 +276,14 @@ class ResidualBasedEliminationBuilderAndSolver
                     }
                 }
 
-		double stop_build = OpenMPUtils::GetCurrentTime();
-		if (this->GetEchoLevel() > 1 && r_model_part.GetCommunicator().MyPID() == 0)
-			std::cout << "build time: " << stop_build - start_build << std::endl;
+      double stop_build = OpenMPUtils::GetCurrentTime();
+      if (this->GetEchoLevel() > 1 && r_model_part.GetCommunicator().MyPID() == 0)
+         std::cout << "build time: " << stop_build - start_build << std::endl;
```
Comment (RiccardoRossi) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]: good place for using info

**Comment by RiccardoRossi** [comment]
I am not against this commit, however did u benchmark the Build times (particularly the ConstructMatrix times) before and after the change? before accepting we need to be sure it doesn't affect the performance of a critical part of the code

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
blocking while waiting for benchmark

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h (hunk 2 above)
Comment (philbucher) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]: @loumalouomega I recommend doing what I did in #1962 regarding the timer stuff

Comment (loumalouomega) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]: In these lines I just changed the "\t" to "    " but OK

**Comment by loumalouomega** [comment]
The times are similar, I ran several times a problem on 200K elements and these are the times for constructing the system:
imagen

**Comment by roigcarlo** [comment]
@RiccardoRossi Don't see anything wrong, I think we can approve

**Review by roigcarlo — DISMISSED** [review]
@loumalouomega You broke something in the residual_based_builder_and_solver.h

**Comment by loumalouomega** [comment]
Strange I not just run the test, but I run the comparisons

**Comment by loumalouomega** [comment]
Last minute type, one sec

**Comment by RiccardoRossi** [comment]
ok, the benchmark is favorable. I am dismissing my previous review and let @roigcarlo approve

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**